### PR TITLE
Refactor Path, Waypoint, Spline using built-in java data structures.

### PR DESCRIPTION
**Pull request by kjrobrien**

### CODE REVIEW

File: src/main/java/edu/wpi/first/pathui/ISpline.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package edu.wpi.first.pathui;
+
+import javafx.scene.Group;
+
+public interface ISpline {
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/pathui/ISpline.java]: Rename to just `Spline` and add documentation

File: src/main/java/edu/wpi/first/pathui/QuickSpline.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    update();
+  }
+
+  @Override
+  public void removeFromGroup(Group splineGroup) {
+    splineGroup.getChildren().remove(curve);
+  }
+
+  /**
+   * Creates a Spline that is "Quick", i.e. a JavaFX CubicCurve.
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/pathui/QuickSpline.java]: The definition of a "quick" spline should go in the class-level docs, not the constructor. Also needs clarification on what a "slow" spline would be

File: src/main/java/edu/wpi/first/pathui/QuickSpline.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    this.end = end;
+    curve = new CubicCurve();
+    curve.setOnDragDetected(event -> {
+      Dragboard board = curve.startDragAndDrop(TransferMode.ANY);
+      board.setContent(Map.of(DataFormats.SPLINE, "Spline"));
+      DragHandler.currentSpline = this;
+    });
+  }
+
+  // Bezier curve to hermite curve
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/pathui/QuickSpline.java]: Should be a JavaDoc comment

File: src/main/java/edu/wpi/first/pathweaver/Path.java [hunk 4]
```diff
@@ -44,7 +45,7 @@ public String toString() {
   }
 
   /**
-   * Creates new Waypoint in Path after previous.
+   * Create new Waypoint in Path after previous.
```
Comment (AustinShalit) [inline src/main/java/edu/wpi/first/pathweaver/Path.java]: Creates

Comment (kjrobrien) [reply src/main/java/edu/wpi/first/pathweaver/Path.java]: Whoops, chose the wrong line in a merge conflict. Should be fixed now.

File: src/main/java/edu/wpi/first/pathweaver/Path.java [hunk 5]
```diff
@@ -1,126 +1,97 @@
...
+   * @param previous The node before this one
+   * @param position Position to play new Waypoint
+   * @param tangent  Tangent vector at the new point
    *
    * @return new Waypoint
    */
-  public Waypoint addNewWaypoint(Waypoint previous, Waypoint next) {
-    if (previous.getNextWaypoint() != next || next.getPreviousWaypoint() != previous) {
-      throw new IllegalArgumentException("New Waypoint not between connected points");
+  public Waypoint addNewWaypoint(Waypoint previous, Point2D position, Point2D tangent, Boolean locked) {
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/pathweaver/Path.java]: Use `boolean` primitive. `Boolean` should _only_ be used when working with generics where primitives are not supported

**Pull request merged**